recipes-extended/xen/files/libxl-add-ovmf-varstored.patch

```diff
@@ -0,0 +1,202 @@
+--- a/tools/libxl/libxl_dm.c
++++ b/tools/libxl/libxl_dm.c
+@@ -21,6 +21,7 @@
+ 
+ #include <xc_dom.h>
+ #include <xen/hvm/e820.h>
++#include <sys/stat.h>
+ #include <sys/types.h>
+ #include <pwd.h>
+ #include <grp.h>
+@@ -2192,6 +2193,45 @@ static void fork_helper(libxl__gc *gc, c
+                          "%d", pid);
+ }
+ 
++static void start_varstored_helper(libxl__gc *gc, const char *helper_path,
++                             int guest_domid, const char *guest_uuid)
++{
++    pid_t pid;
++    libxl_ctx *ctx = libxl__gc_owner(gc);
++
++    struct stat st = {0};
++    if (stat(GCSPRINTF("/var/run/varstored-root-%d", guest_domid), &st) == -1) {
++        mkdir(GCSPRINTF("/var/run/varstored-root-%d", guest_domid), 0700);
++    }
++
++    pid = fork();
++    if (pid == -1) {
++        LOG(ERROR, "Failed to fork");
++    } else if (pid == 0) {
++        execl(helper_path, GCSPRINTF("varstored-watch-%d", guest_domid),
++                       "--domid",
++                        GCSPRINTF("%d", guest_domid),
++                        "--uuid",
++                        GCSPRINTF("%s", guest_uuid),
++                        NULL);
++    }
++    //Wait for a PID from varstored to be created.
++    unsigned int timeout;
++    char *ready = NULL;
++    const char *varstored_path =
++        GCSPRINTF("%s/varstored-pid",
++                  libxl__xs_get_dompath(gc, guest_domid));
++    for (timeout = 0; !ready && timeout < 4; ++timeout) {
++        ready = xs_read(ctx->xsh, XBT_NULL, varstored_path, NULL);
++        if (ready) {
++            free(ready);
++            break;
++        }
++        sleep(1);
++    }
++
++}
++
+ void libxl__spawn_stub_dm(libxl__egc *egc, libxl__stub_dm_spawn_state *sdss)
+ {
+     STATE_AO_GC(sdss->dm.spawn.ao);
+@@ -2349,6 +2389,16 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+         fork_helper(gc, AUDIO_HELPER_PATH, XS_AUDIO_PID, guest_domid, dm_domid);
+     }
+ 
++    if (guest_config->b_info.u.hvm.bios == LIBXL_BIOS_TYPE_OVMF) {
```

Comment:
I thought this would check libxl_defbool_val(info->u.hvm.secureboot) to skip starting the watcher/varstored.  Does that still work?

Reply:
The varstore is not dependent on secureboot, there are a bunch of other EFI vars that it tracks (eg BootOrder, PlatformLang, MemoryTypeInformation, etc). It should be started for all EFI guests.